src/compiler.ts

```diff
@@ -8682,12 +8682,36 @@ export class Compiler extends DiagnosticEmitter {
     var values = expression.values;
     var members = classReference.members;
     var hasErrors = false;
-    var exprs = new Array<ExpressionRef>(numNames + 2);
+    var exprs = new Array<ExpressionRef>();
     var flow = this.currentFlow;
     var tempLocal = isManaged
       ? flow.getAutoreleaseLocal(classReference.type)
       : flow.getTempLocal(classReference.type);
     assert(numNames == values.length);
+
+    // Assume all class fields will be omitted, and add them to our omitted list
+    var omittedClassFieldMembers = new Map();
+    if (members) {
+      for (let memberKey of members.keys()) {
+        let member = members.get(memberKey);
+        if (member && member.kind == ElementKind.FIELD) {
```

Comment:
```suggestion
        if (member !== null && member.kind == ElementKind.FIELD) {
```
That's necessary due to some obscurities when compiling logical expressions. This way, all parts are `bool`.